src/main/java/com/example/demo/controller/CategoryController.java

```diff
@@ -0,0 +1,88 @@
+package com.example.demo.controller;
+
+import com.example.demo.dto.book.BookDtoWithoutCategoryIds;
+import com.example.demo.dto.category.CategoryResponseDto;
+import com.example.demo.dto.category.CreateCategoryRequestDto;
+import com.example.demo.service.book.BookService;
+import com.example.demo.service.category.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for managing categories")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @GetMapping
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Get all categories",
+            description = "Get a list of all available categories")
+    public List<CategoryResponseDto> getAll(Pageable pageable) {
```

Comment:
better return Page. How you can iterate by pages if you don’t  now count of pages